v2/elasticsearch-common/src/main/java/com/google/cloud/teleport/v2/elasticsearch/options/ElasticsearchWriteOptions.java

```diff
@@ -22,24 +22,28 @@
 
 /** The {@link ElasticsearchWriteOptions} with the common write options for Elasticsearch. * */
 public interface ElasticsearchWriteOptions extends PipelineOptions {
-  @Description(
-      "Comma separated list of Elasticsearch nodes to connect to, ex: http://my-node1,http://my-node2")
+  @Description("Elasticsearch URL in format http://hostname:[port] or Base64 encoded CloudId")
   @Validation.Required
-  String getTargetNodeAddresses();
+  String getConnectionUrl();
 
-  void setTargetNodeAddresses(String targetNodeAddresses);
+  void setConnectionUrl(String connectionUrl);
 
-  @Description("The index toward which the requests will be issued, ex: my-index")
+  @Description("Username for Elasticsearch endpoint")
   @Validation.Required
-  String getWriteIndex();
+  String getWriteElasticsearchUsername();
 
-  void setWriteIndex(String writeIndex);
+  void setWriteElasticsearchUsername(String writeElasticsearchUsername);
 
-  @Description("The document type toward which the requests will be issued, ex: my-document-type")
+  @Description("Password for Elasticsearch endpoint")
   @Validation.Required
-  String getWriteDocumentType();
+  String getWriteElasticsearchPassword();
+
+  void setWriteElasticsearchPassword(String writeElasticsearchPassword);
+
+  @Description("The index toward which the requests will be issued, ex: my-index")
+  String getWriteIndex();
 
-  void setWriteDocumentType(String writeDocumentType);
+  void setWriteIndex(String writeIndex);
```

Comment:
> `getWriteIndex` and `setWriteIndex` to `getIndex`and `setWriteIndex`

To add context, reason behind write/read prefixes for options is to support elasticsearch-to-elasticsearch template scenario where users may need to provide different connection url / credentials for source vs target cluster.

Reply:
@suyograo @prathapreddy123 
To make it work we should rename both getters and setters, otherwise we'll get IllegalArgumentException.

Reply:
> @suyograo @prathapreddy123
> To make it work we should rename both getters and setters, otherwise we'll get IllegalArgumentException.

I would recommend to continue with prefixes without any changes.